pkg/types/azure/validation/machinepool.go

```diff
@@ -12,5 +12,14 @@ func ValidateMachinePool(p *azure.MachinePool, fldPath *field.Path) field.ErrorL
 	if p.OSDisk.DiskSizeGB < 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("diskSizeGB"), p.OSDisk.DiskSizeGB, "Storage DiskSizeGB must be positive"))
 	}
+
+	if p.OSDisk.DiskType != "" {
+		//"UltraSSD_LRS": "",
+		_, found := map[string]string{"Standard_LRS": "", "StandardSSD_LRS": "", "Premium_LRS": ""}[p.OSDisk.DiskType]
```

Comment:
nit:
keep a list of known valid values and then run `sets.NewString(knownDiskTypes...).Has(p.OSDisk.DiskType)`

that helps with readability as a constant list is easy to track and maintain outside the function.